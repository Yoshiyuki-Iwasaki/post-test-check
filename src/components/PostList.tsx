import React, { useEffect,useState } from 'react'
import PostItem from './PostItem'
import PostInput from "./PostInput";
import styled from "styled-components";
import data from '../data.json';

const Main = styled.ul`
  margin: 20px auto 0;
  background: #fff;
  height: 600px;
  overflow: scroll;
}
`;

const PostList = () => {
  const [postDate, setPostDate] = useState<any>([]);

  useEffect(() => {
    setPostDate(data);
  }, []);

  return (
    <>
      <PostInput
        postDate={postDate}
        setPostDate={setPostDate}
      />
      <Main>
        {postDate &&
          postDate.map((data: any, index: number) => (
            <PostItem key={index} data={data} />
          ))}
      </Main>
    </>
  );
}

export default PostList
