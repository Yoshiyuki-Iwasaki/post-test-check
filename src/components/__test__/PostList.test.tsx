import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import PostList from "../PostList";

describe("PostList 機能テスト", () => {
  afterEach(() => {
    cleanup();
  });

  // Snapshot テスト
  test("render", () => {
    const { asFragment } = render(<PostList />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Submit テスト
  test("Submit テスト", () => {
    const component = render(<PostList />);
  });
});
