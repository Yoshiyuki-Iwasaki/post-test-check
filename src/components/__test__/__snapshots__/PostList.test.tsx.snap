// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostList> Snapshotは問題ないか 1`] = `
<DocumentFragment>
  <form
    class="sc-jrQzAO jAVXgP"
  >
    <input
      class="sc-kDTinF hQVfki"
      type="text"
      value=""
    />
  </form>
  <ul
    class="sc-iqseJM cytntT"
  >
    <li
      class="sc-gsDKAQ cAoxjG"
    >
      <div
        class="sc-dkPtRN hvIKIu"
      >
        <div
          class="sc-hKwDye iqWrVV"
        >
          <span
            class="sc-eCImPb ezjWWd"
          />
        </div>
        <div
          class="sc-jRQBWg elvELz"
        >
          <div
            class="sc-gKclnd fVLuFs"
          >
            <p
              class="sc-iCfMLu kTZGvT"
            >
              username1
            </p>
            <p
              class="sc-furwcr eSrJdJ"
            >
              date1
            </p>
          </div>
          <p
            class="sc-pVTFL djKbWE"
          >
            投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容
          </p>
        </div>
      </div>
      <button
        class="sc-bdvvtL caPMGC"
        data-testid="like"
      >
        0
      </button>
    </li>
    <li
      class="sc-gsDKAQ cAoxjG"
    >
      <div
        class="sc-dkPtRN hvIKIu"
      >
        <div
          class="sc-hKwDye iqWrVV"
        >
          <span
            class="sc-eCImPb ezjWWd"
          />
        </div>
        <div
          class="sc-jRQBWg elvELz"
        >
          <div
            class="sc-gKclnd fVLuFs"
          >
            <p
              class="sc-iCfMLu kTZGvT"
            >
              username2
            </p>
            <p
              class="sc-furwcr eSrJdJ"
            >
              date2
            </p>
          </div>
          <p
            class="sc-pVTFL djKbWE"
          >
            投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容
          </p>
        </div>
      </div>
      <button
        class="sc-bdvvtL caPMGC"
        data-testid="like"
      >
        0
      </button>
    </li>
    <li
      class="sc-gsDKAQ cAoxjG"
    >
      <div
        class="sc-dkPtRN hvIKIu"
      >
        <div
          class="sc-hKwDye iqWrVV"
        >
          <span
            class="sc-eCImPb ezjWWd"
          />
        </div>
        <div
          class="sc-jRQBWg elvELz"
        >
          <div
            class="sc-gKclnd fVLuFs"
          >
            <p
              class="sc-iCfMLu kTZGvT"
            >
              username3
            </p>
            <p
              class="sc-furwcr eSrJdJ"
            >
              date3
            </p>
          </div>
          <p
            class="sc-pVTFL djKbWE"
          >
            投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容投稿内容
          </p>
        </div>
      </div>
      <button
        class="sc-bdvvtL caPMGC"
        data-testid="like"
      >
        0
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
